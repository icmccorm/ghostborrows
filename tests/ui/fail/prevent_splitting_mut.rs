use ghostborrows::*;

fn main() {
    let value = Value::new(1);
    value.borrow_mut(|r1, token1| {
        let activated = r1.activate(token1);
        activated.borrow_mut(|_, _| {
            let (_, _) = r1.split();
        });
    });
}