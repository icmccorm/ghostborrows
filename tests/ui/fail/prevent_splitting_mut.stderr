error[E0382]: use of moved value: `r1`
 --> tests/ui/fail/prevent_splitting_mut.rs:7:30
  |
5 |     value.borrow_mut(|r1, token1| {
  |                       -- move occurs because `r1` has type `RefReserved<'_, i32>`, which does not implement the `Copy` trait
6 |         let activated = r1.activate(token1);
  |                            ---------------- `r1` moved due to this method call
7 |         activated.borrow_mut(|_, _| {
  |                              ^^^^^^ value used here after move
8 |             let (_, _) = r1.split();
  |                          -- use occurs due to use in closure
  |
note: `RefReserved::<'tag, T>::activate` takes ownership of the receiver `self`, which moves `r1`
 --> src/refs.rs
  |
  |     pub fn activate(self, _token: Token<'tag, T>) -> RefMut<'tag, T> {
  |                     ^^^^
